pkg/util/k8sutil/pod_util.go

```diff
@@ -67,6 +68,26 @@ func containerWithRequirements(c v1.Container, r v1.ResourceRequirements) v1.Con
 	return c
 }
 
+func etcdReadinessProbe(isSecure bool) *v1.Probe {
+	// etcd pod is ready only if a linearizable get succeeds.
+	cmd := "ETCDCTL_API=3 etcdctl get foo"
```

Comment:
Unify this with liveness probe code?

Reply:
The initial delay, timeout, period, and failure threshold are smaller for readiness probe than they are for liveness probe. 
Or do you still want to unify them into a common function and specify those values are params?

Reply:
Compose them:
```
pb := NewProber(xxx)
SetProberInitialDelay(pb, 5)
...
```